ir/src/main/scala/org/scalajs/ir/ScalaJSVersions.scala

```diff
@@ -12,32 +12,94 @@
 
 package org.scalajs.ir
 
-object ScalaJSVersions {
+import java.util.concurrent.ConcurrentHashMap
 
-  /* DO NOT MAKE THESE 'final val's!
-   * When referring to these "constants" from separate libraries, if it is a
-   * 'final val', the value will be copied in the binaries of those libraries.
-   * If they are then linked to a different version of the IR artifact, their
-   * copy of these constants will not be updated.
-   */
+import scala.util.matching.Regex
 
+object ScalaJSVersions extends VersionChecks(
   /** Scala.js version. */
-  val current: String = "1.0.0-SNAPSHOT"
-
-  /** Version of binary IR emitted by this version of Scala.js.
-   *
-   *  This should be either of:
-   *  - a prior release version (i.e. "0.5.0", *not* "0.5.0-SNAPSHOT")
-   *  - `current`
-   */
-  val binaryEmitted: String = current
-
-  /** Versions whose binary files we can support (used by deserializer) */
-  val binarySupported: Set[String] = {
-    Set(binaryEmitted)
+  current = "1.0.0-SNAPSHOT",
+  /** Version of binary IR emitted by this version of Scala.js. */
+  binaryEmitted = "1.0-SNAPSHOT"
```

Comment:
These four lines should have 4 spaces of indentation.

Reply:
And the Scaladoc comments should be on the `val`s in `VersionChecks`. Actual arguments don't care about Scaladoc comments.